### DISCUSSION TITLE
[TypeScript] is there a plan to migrate to the type declaration-based `defineProps()`

### DISCUSSION THREAD

**Discussion by ToyCat93** [element-plus/element-plus#17788] [Feature Request]
Feature type: **`TypeScript`**

## Details

- Enhanced component name: `All`

## Description

when I want to define a `const` of type `InputProps`, IDEA will prompt me that all necessary properties need to be declared, even if a certain property is not necessary.

for example:

```ts
import type { InputProps } from 'element-plus'

export const defaultProp: InputProps = {
  type: 'textarea',
}
```

IDEA will throw this message: 
```
type { type: string; } is missing the following properties from type InputProps: 
disabled, modelValue, autosize, autocomplete, and 8 more.
```
can we improve it by migrating to type declaration-based `defineProps()`?

for example:

```ts 
// input.ts

export type InputProps = {
  id?: string
  size?: '' | 'default' | 'small' | 'large'
  disabled?: boolean
  modelValue?: string | number | null
  maxlength?: string | number
  minlength?: string | number
  type?: string
  // more...
}
```

```vue
// input.vue

<script lang="ts" setup>
import { InputProps } from './input'

const { type = 'text' } = defineProps<InputProps>()
</script>
```

the above, is it feasible?  

### Additional comments

reference: https://vuejs.org/api/sfc-script-setup.html#type-only-props-emit-declarations

**Comment by tuskermanshu** [comment]
It is recommended to use the ts generalization tools  `Pick` or `Omit` to implement the
